File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/managetaskview/IconSpinnerItem.java
package com.wiseassblog.samsaradayplanner.ui.managetaskview;

import android.content.Context;

import com.wiseassblog.samsaradayplanner.common.IconUtility;
import com.wiseassblog.samsaradayplanner.domain.constants.ICON;

import java.util.ArrayList;
import java.util.List;

class IconSpinnerItem {

    public static final int NUMBER_OF_ICONS = 9;

    public static List<IconSpinnerItem> getItems(Context context) {
        List<IconSpinnerItem> list = new ArrayList<>();

        for (int index = 0; index < NUMBER_OF_ICONS; index++) {
            list.add(
                    new IconSpinnerItem(
                            IconUtility.iconNames[index],
                            IconUtility.getResIdFromEnum(
                                    context,
                                    ICON.values()[index]
                            )
                    )
            );
        }

        return list;
    }

    private final String itemText;
    private final int itemIconResId;

    public IconSpinnerItem(String itemText, int itemIconResId) {
        this.itemText = itemText;
        this.itemIconResId = itemIconResId;
    }

    public String getItemText() {
        return itemText;
    }

    public int getItemIconResId() {
        return itemIconResId;
    }
}

File: app/src/main/java/com/wiseassblog/samsaradayplanner/ui/managetaskview/IconSpinnerItem.kt
package com.wiseassblog.samsaradayplanner.ui.managetaskview

import android.content.Context
import com.wiseassblog.samsaradayplanner.common.getResIdFromEnum
import com.wiseassblog.samsaradayplanner.domain.constants.ICON
import com.wiseassblog.samsaradayplanner.domain.constants.ICON_NAMES
import java.util.*

internal class IconSpinnerItem(val itemText: String, val itemIconResId: Int) {

    companion object {
        const val NUMBER_OF_ICONS = 9

        @JvmStatic
        fun getItems(context: Context?): List<IconSpinnerItem> {
            val list: MutableList<IconSpinnerItem> = ArrayList()
            for (index in 0 until NUMBER_OF_ICONS) {
                list.add(
                    IconSpinnerItem(
                        ICON_NAMES.get(index),
                        getResIdFromEnum(
                            context!!,
                            ICON.values()[index]
                        )
                    )
                )
            }
            return list
        }
    }
}